.circleci/config.yml

```diff
@@ -822,15 +822,19 @@ jobs:
       - run:
           name: Compatibility Integration Tests
           command: |
-            subtests=$(ls -d test/integration/consul-container/*/ | grep -v libs | xargs -n 1 basename | circleci tests split)
-            echo "Running $(echo $subtests | wc -w) subtests"
-            echo "$subtests"
-            subtests_formatted=$(echo "$subtests" |  sed 's/^/.\//g' | xargs)
             mkdir -p /tmp/test-results/
-            docker run consul:local consul version
             cd ./test/integration/consul-container
-            echo "$subtests_formatted"
-            gotestsum -- -timeout=30m ./$subtests_formatted --target-version local --latest-version latest
+            docker run --rm consul:local consul version
+            gotestsum \
+              --format=short-verbose \
+              --debug \
+              --rerun-fails=3 \
```

Comment:
In order to rerun failures and also pass args to `go test` (like timeout, target-version, etc) the packages need to be specified with `--packages`. For the custom args like target-version to work the need to come after the package placement, so we just avoid doing test splits here and specify `./...` twice to make it work at all.